### PR TITLE
ci: use self-hosted for amd

### PR DESCRIPTION
**Pull request by ChrisKujawa**
## Description

Use for the smoke linux AMD job OUR self-hosted runners instead of the infra provided ones, since they seem to have a long delay which caused bors to time out.

Affected PR's:

 * https://github.com/camunda/zeebe/pull/11281
 * https://github.com/camunda/zeebe/pull/11313
 * https://github.com/camunda/zeebe/pull/11296

## Related issues

closes #

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

**Review by korthout — APPROVED** [review]
OMG Thanks ❤️ 

bors merge

**Comment by korthout** [comment]
bors p=1

**Comment by megglos** [comment]
@Zelldon there are no 4 core amd64 runners in our self-hosted setup yet.
You would need to add a pool for them in our GKE cluster + setup the runner and scaler in https://github.com/zeebe-io/infra/tree/main/gcp/zeebe-io/zeebe-ci-1

Alternatively, let's try the Github hosted `ubuntu-latest` here, the experienced some occasional maven timeouts in the past but that may still be better than timeouts.

**Comment by Zelldon** [comment]
Can also use thr other nodes with 16 CPU?

**Comment by megglos** [comment]
and there goes the 💸 😆 

but yeah we can optimize it by adding a 4 or 8 pool next year.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -189,7 +189,7 @@ jobs:
           - os: windows
             runner: windows-latest
           - os: linux
-            runner: n1-standard-8-netssd-preempt
+            runner: [ self-hosted, linux, amd64, "4" ]
```
Comment (megglos) [inline .github/workflows/ci.yml:192]: you may just define it like this, any core count runner should be used then going forward
```suggestion
            runner: [ self-hosted, linux, amd64 ]
```

**Comment by Zelldon** [comment]
Is it taking the whole node ? Or why I need to add a new node pool? If this is the case I get why it is more expensive.

**Comment by megglos** [comment]
> Is it taking the whole node ? Or why I need to add a new node pool? If this is the case I get why it is more expensive.

yes a runner always takes a whole node right now

File: .github/workflows/ci.yml (hunk 1 above)
Comment (Zelldon) [reply .github/workflows/ci.yml:192]: Thanks :+1:

**Comment by Zelldon** [comment]
bors p=1

**Pull request merged**